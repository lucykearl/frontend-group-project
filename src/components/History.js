import React from "react";
import JsonData from "../data/history.json";


function History(){
  const DisplayData=JsonData.map(
      (info)=>{
          return(
              <tr>
                  <td>{info.date}</td>
                  <td>{info.time}</td>
                  <td>{info.exercises.map((exercise) => exercise.title + ", ")}</td>
                  <td>{info.exercises.map((exercise) => exercise.weight + ", ")}</td>
                  <td>{info.exercises.map((exercise) => exercise.sets + ", ")}</td>
                  <td>{info.exercises.map((exercise) => exercise.reps + ", ")}</td>
              </tr>
          )
      }
  )

  return(
      <div>
          <table class="table table-striped">
              <thead>
                  <tr>
                  <th>Date</th>
                  <th>Time</th>
                  <th>Title</th>
                  <th>Weight</th>
                  <th>Sets</th>
                  <th>Reps</th>
                  </tr>
              </thead>
              <tbody>
               
                  
                  {DisplayData}
                  
              </tbody>
          </table>
           
      </div>
  )
}

export default History;
