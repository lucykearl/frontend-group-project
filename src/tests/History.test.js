import { render, screen } from "@testing-library/react";
import History from "../components/History";



describe("history", () => {
    const validProps = [
        {
            "Date": "21/10/2022", "Time": "51:42", "Exercises": [{
                "title": "Bicep Curl",
                "weight": "10kg",
                "sets": 4,
                "reps": 10
            }
            ]
        }
    ]

    test("the component renders", () => {
        const { asFragment } = render(<History />);

        expect(asFragment()).toMatchSnapshot();
    });

    test("it renders a table", () => {
        render (<History />);
        expect(screen.findByRole("table")).toBeInTheDocument();
    }
    );
    // // it("renders correct values for props", () => {
    //     const { getByText, getByTestId } = render(
    //         <ForecastSummary
    //           date={validProps.date}
    //           description={validProps.description}
    //           icon={validProps.icon}
    //           temperature={validProps.temperature}
    //           onSelect={validProps.onSelect}
    //         />
    //       );//


    test("it should render date", () => {
        const { getByText } = render(<History  />)
        expect(getByText("21/10/2022")).toHaveClass("history--date")
    });

    xtest("it should return a title", () => {
        const { getAllByText } = render(<History />)
        expect(getAllByText("Bicep Curl, Hammer Curl, Skull Crusher, Upright Row, ", 
         )).toHaveClass("history--exercise--title")
    });

   


    test("it should return a weight", () => { });


    test("it should return the number of sets", () => { });

    test("it should return the number of reps", () => { });
});
