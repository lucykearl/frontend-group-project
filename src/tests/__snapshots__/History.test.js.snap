// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`history the component renders 1`] = `
<DocumentFragment>
  <div
    class="history-table"
  >
    <table
      class="table table-striped"
      id="historytable"
    >
      <thead>
        <tr>
          <th>
            Date
          </th>
          <th>
            Time
          </th>
          <th>
            Title
          </th>
          <th>
            Weight
          </th>
          <th>
            Reps
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</DocumentFragment>
`;
